soh/src/code/z_message_PAL.c

```diff
@@ -1673,9 +1673,73 @@ void Message_OpenText(GlobalContext* globalCtx, u16 textId) {
                             //font->msgLength, __FILE__, __LINE__);
     } else {
         Message_FindMessage(globalCtx, textId);
-        msgCtx->msgLength = font->msgLength;
-        char* src = (uintptr_t)font->msgOffset;
-        memcpy(font->msgBuf, src, font->msgLength);
+        // if we're rando'd and talking to a gossip stone
+        if (gSaveContext.n64ddFlag &&
+            textId == 0x2053 &&
+            GetRandoSettingValue(RSK_GOSSIP_STONE_HINTS) != 0 &&
```

Comment:
for example here.. would be a very nice place to have used an extracted checker method.

Reply:
made an issue for it and added it to the rando v2 milestone https://github.com/HarbourMasters/Shipwright/issues/741